src/OrchardCore.Modules/OrchardCore.Themes/Controllers/AdminController.cs

```diff
@@ -129,7 +125,7 @@ public class AdminController : Controller
                 return Unauthorized();
             }
 
-            var feature = _extensionManager.GetFeatures().FirstOrDefault(f => f.Extension.Manifest.IsTheme() && f.Id == id);
+            var feature = _extensionManager.GetFeatures().FirstOrDefault(f => f.Extension is IThemeExtensionInfo && f.Id == id);
```

Comment:
How is that better?

Reply:
@sebastienros 

Humm, i think because now we can directly check the extension type without having to call an helper on the extension manifest which do this.

    return manifestInfo?.Type?.Equals("theme", StringComparison.OrdinalIgnoreCase) ?? false;

 I thought about creating an helper on the extension class directly but `extension.IsTheme()` would just do `return extension is IThemeExtensionInfo`, so ...

**But it is as you want**.

Note: And because i saw the comments just before from @sfmskywalker.

    // TODO: Cast the IExtensionInfo objects to ThemeExtensionInfo objects once Nick fixes the issue where both Modules and Themes are constructed as ExtensionInfos.
    // var themes = _extensionManager.GetExtensions().OfType<ThemeExtensionInfo>().Where(extensionDescriptor =>
     var themes = _extensionManager.GetExtensions().Where(extensionDescriptor =>
    {
         var isTheme = extensionDescriptor.Manifest.IsTheme();